azure/Guidelines.md

```diff
@@ -0,0 +1,889 @@
+
+:white_check_mark: **DO** return a `204-No Content` without a resource/body for a DELETE operation (even if the URL identifies a resource that does not exist; do not return `404-Not Found`)
+
+:white_check_mark: **DO** return a `403-Forbidden` when the user does not have access to the resource _unless_ this would leak information about the existence of the resource that should not be revealed for security/privacy reasons, in which case the response should be `404-Not Found`. [Rationale: a `403-Forbidden` is easier to debug for customers, but should not be used if even admitting the existence of things could potentially leak customer secrets.]
+
+:white_check_mark: **DO** support caching and optimistic concurrency by honoring the the `If-Match`, `If-None-Match`, if-modified-since, and if-unmodified-since request headers and by returning the ETag and last-modified response headers
+
+### HTTP Query Parameters and Header Values
+Because information in the service URL, as well as the request / response, are strings, there must be a predictable, well-defined scheme to convert strings to their corresponding values.
+
+:white_check_mark: **DO** validate all query parameter and request header values and fail the operation with `400-Bad Request` if any value fails validation. Return an error response as described in the [Handling Errors](#Handling-errors) section indicating what is wrong so customer can diagnose the issue and fix it themselves.
+
+:white_check_mark: **DO** use the following table when translating strings:
+
+Data type | Document that string must be
+--------- | -------
+Boolean   | true / false (all lowercase)
+Integer   | -2<sup>53</sup>+1 to +2<sup>53</sup>-1 (for consistency with JSON limits on integers [RFC8259](https://datatracker.ietf.org/doc/html/rfc8259))
+Float     | [IEEE-754 binary64](https://en.wikipedia.org/wiki/Double-precision_floating-point_format)
+String    | (Un)quoted?, max length, legal characters, case-sensitive, multiple delimiter
+UUID      | 123e4567-e89b-12d3-a456-426614174000 (no {}s, hyphens, case-insensitive) [RFC4122](https://datatracker.ietf.org/doc/html/rfc4122)
+Date/Time (Header) | Sun, 06 Nov 1994 08:49:37 GMT [RFC7231](https://datatracker.ietf.org/doc/html/rfc7231#section-7.1.1.1)
+Date/Time (Query parameter) | YYYY-MM-DDTHH:mm:ss.sssZ (with at most 3 digits of fractional seconds) [RFC3339](https://datatracker.ietf.org/doc/html/rfc3339)
+Byte array | Base-64 encoded, max length
+Array      | One of a) a comma-separated list of values (preferred), or b) separate `name=value` parameter instances for each value of the array
+
+
+The table below lists the headers most used by Azure services:
+
+Header Key          | Applies to | Example
+------------------- | ---------- | -------------
+_authorization_     | Request    | Bearer eyJ0...Xd6j (Support Azure Active Directory)
```

Comment:
Should "Azure Active Directory" be a link to the information on how to do this?

Reply:
Ideally yes. I don't have that link thought. Scott Schaab might have it.